### PR TITLE
Almost ready

### PR DESCRIPTION
**Pull request by BillyGalbreath**
I skipped this patch. Didn't know how to resolve the conflicts.
0385-only-add-passanger-entities-once-from-spawners.patch

It still doesn't compile because of MojangAPI 1.16.2 missing dependency.

Enjoy ;)

### CODE REVIEW

File: Spigot-Server-Patches/0407-Entity-Activation-Range-2.0.patch [hunk 1]
```diff
@@ -14,35 +14,43 @@ Adds flying monsters to control ghast and phantoms
...
 --- a/src/main/java/net/minecraft/server/Entity.java
 +++ b/src/main/java/net/minecraft/server/Entity.java
-@@ -166,7 +166,7 @@ public abstract class Entity implements INamableTileEntity, ICommandListener, Ke
-     public boolean ac;
+@@ -163,7 +163,11 @@ public abstract class Entity implements INamableTileEntity, ICommandListener, Ke
+     public boolean Y;
      public boolean impulse;
      public int portalCooldown;
--    protected boolean inPortal; public final boolean inPortal() { return this.inPortal; } // Paper - OBFHELPER
++<<<<<<< HEAD
```
Comment (Spottedleaf) [inline Spigot-Server-Patches/0407-Entity-Activation-Range-2.0.patch]: this needs to be resolved

**Comment by Proximyst** [comment]
Comment hidden; see kashike's comment.

<details>
<summary>Incorrect comment</summary>
Here's the missing patch as far as I understood it :)

```diff
diff --git a/src/main/java/net/minecraft/server/MobSpawnerAbstract.java b/src/main/java/net/minecraft/server/MobSpawnerAbstract.java
index c2e830db7a..e021e5922f 100644
--- a/src/main/java/net/minecraft/server/MobSpawnerAbstract.java
+++ b/src/main/java/net/minecraft/server/MobSpawnerAbstract.java
@@ -186,6 +186,7 @@ public abstract class MobSpawnerAbstract {
                                 return;
                             }

+                            this.addPassengerEntity(entity); // Paper - only add passenger entities once from spawners
                             world.triggerEffect(2004, blockposition, 0);
                             if (entity instanceof EntityInsentient) {
                                 ((EntityInsentient) entity).doSpawnEffect();
@@ -204,6 +205,16 @@ public abstract class MobSpawnerAbstract {
         }
     }

+    // Paper start - only add passenger entities once from spawners
+    private void addPassengerEntity(Entity entity) {
+        if (!entity.valid
+            || !this.getWorld().addEntity(entity, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.SPAWNER)) return;
+        for (Entity passenger : entity.getPassengers()) {
+            this.addPassengerEntity(entity);
+        }
+    }
+    // Paper end
+
     public void resetTimer() { i(); } // Paper - OBFHELPER
     private void i() {
         if (this.maxSpawnDelay <= this.minSpawnDelay) {
@@ -329,6 +340,7 @@ public abstract class MobSpawnerAbstract {

     public abstract void a(int i);

+    public final World getWorld() { return this.a(); } // Paper - OBFHELPER
     public abstract World a();

     public abstract BlockPosition b();
```
</details>

**Comment by kashike** [comment]
> 
> 
> Here's the missing patch as far as I understood it :)
> 
> ```diff
> diff --git a/src/main/java/net/minecraft/server/MobSpawnerAbstract.java b/src/main/java/net/minecraft/server/MobSpawnerAbstract.java
> index c2e830db7a..e021e5922f 100644
> --- a/src/main/java/net/minecraft/server/MobSpawnerAbstract.java
> +++ b/src/main/java/net/minecraft/server/MobSpawnerAbstract.java
> @@ -186,6 +186,7 @@ public abstract class MobSpawnerAbstract {
>                                  return;
>                              }
> 
> +                            this.addPassengerEntity(entity); // Paper - only add passenger entities once from spawners
>                              world.triggerEffect(2004, blockposition, 0);
>                              if (entity instanceof EntityInsentient) {
>                                  ((EntityInsentient) entity).doSpawnEffect();
> @@ -204,6 +205,16 @@ public abstract class MobSpawnerAbstract {
>          }
>      }
> 
> +    // Paper start - only add passenger entities once from spawners
> +    private void addPassengerEntity(Entity entity) {
> +        if (!entity.valid
> +            || !this.getWorld().addEntity(entity, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.SPAWNER)) return;
> +        for (Entity passenger : entity.getPassengers()) {
> +            this.addPassengerEntity(entity);
> +        }
> +    }
> +    // Paper end
> +
>      public void resetTimer() { i(); } // Paper - OBFHELPER
>      private void i() {
>          if (this.maxSpawnDelay <= this.minSpawnDelay) {
> @@ -329,6 +340,7 @@ public abstract class MobSpawnerAbstract {
> 
>      public abstract void a(int i);
> 
> +    public final World getWorld() { return this.a(); } // Paper - OBFHELPER
>      public abstract World a();
> 
>      public abstract BlockPosition b();
> ```

No.

```java

                            if (!worldserver.addAllEntitiesSafely(entity, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.SPAWNER)) { // CraftBukkit
                                this.i();
                                return;
                            }
```
This code replaced that method.

File: Spigot-Server-Patches/0491-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch [hunk 2]
```diff
@@ -628,29 +628,29 @@ index b419f0e938b8406874ff7b3d0ee7a532e5990557..87810f007ad258935aa54c0f6a2e2d43
...
 +++ b/src/main/java/net/minecraft/server/PlayerChunk.java
-@@ -26,8 +26,8 @@ public class PlayerChunk {
+@@ -28,8 +28,8 @@ public class PlayerChunk {
      private CompletableFuture<IChunkAccess> chunkSave;
      public int oldTicketLevel;
      private int ticketLevel;
 -    private int n;
 -    final ChunkCoordIntPair location; // Paper - private -> package
-+    volatile int n; public final int getCurrentPriority() { return n; } // Paper - OBFHELPER - make volatile since this is concurrently accessed
++    private int n; public final int getCurrentPriority() { return n; } // Paper - OBFHELPER - make volatile since this is concurrently accessed
```
Comment (Spottedleaf) [inline Spigot-Server-Patches/0491-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch]: volatile

**Comment by Proximyst** [comment]
Billy (should've >:() went to sleep, so I'll note here that it runs: https://github.com/BillyGalbreath/Paper/pull/1

**Pull request merged**